app/service/pic-service.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+module.exports = ['$q', '$log', '$http', 'Upload', 'authService', picService];
+
+function picService($q, $log, $http, Upload, authService) {
+  $log.debug('picService');
+
+  let service = {};
+
+  service.uploadGalleryPic = function(galleryData, picData) {
+    $log.debug('uploadGalleryPic');
+
+    return authService.getToken()
+    .then( token => {
+      let url = `${__API_URL__}/api/gallery/${galleryData._id}/pic`;
+      let headers = {
+        Authorization: `Bearer ${token}`,
+        Accept: 'application/json'
+      };
+
+      return Upload.upload({
+        url,
+        headers,
+        method: 'POST',
+        data: {
+          name: picData.name,
+          desc: picData.desc,
+          file: picData.file
+        }
+      });
+    })
+    .then( res => {
+      galleryData.pics.unshift(res.data);
+      return res.data;
+    })
+    .catch( err => {
+      $log.error(err.message);
+      return $q.reject(err);
+    });
+  };
+
+  service.deletePic = function(galleryData, picData) {
+    $log.debug('picService.deletePic');
+
+    return authService.getToken()
+    .then( token => {
+      let url = `${__API_URL__}/api/gallery/${galleryData._id}/pic/${picData._id}`;
+      let config = {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        }
+      };
+
+      return $http.delete(url, config);
+    })
+    .then( res => {
+      for (let i = 0; i < galleryData.pics.length; i++) {
```

Comment:
This is a pretty optimal place to use the `.filter()` function. Not necessary but I thought I would let you know. Functional programming FTW!!!